network/p2p/topic_validator_test.go

```diff
@@ -73,11 +73,8 @@ func TestTopicValidator_Unstaked(t *testing.T) {
 	_, err = sn2.Subscribe(topic, unittest.NetworkCodec(), unittest.AllowAllPeerFilter(), slashingViolationsConsumer)
 	require.NoError(t, err)
 
-	// assert that the nodes are connected as expected
-	require.Eventually(t, func() bool {
-		return len(sn1.ListPeers(topic.String())) > 0 &&
-			len(sn2.ListPeers(topic.String())) > 0
-	}, 3*time.Second, 100*time.Millisecond)
```

Comment:
Since this was checking just that the peer list was non-empty, wouldn't that account for peers that connected early?

Reply:
No in general the 2 processes of subscribing and discovering peers interested in the topic are async. This discovery and connect process may finish before this peer list is updated or the peer list may not be updated at all in some cases. Thus checking this peer list is unnecessary and causes flakiness.